### ISSUE TITLE
ParallelAccelerator variable assignment in parfor not reflected

### ISSUE THREAD

**Issue by ehsantn** [numba/numba#2854]
The output of this program is `[0. 0. 0.]` but it should be `[1. 1. 1.]`.

```python
import numpy as np
from numba import prange, njit

@njit(parallel=True)
def test_impl(n):
    X = np.ones((n, 3))
    y = np.zeros(3)
    for i in prange(n):
        y = X[i,:]
    return y

n = 211
print(test_impl(n))
```

**Comment by ehsantn** [comment]
@DrTodd13 looks like a threading backend issue since sequential lowering is fine.

**Comment by DrTodd13** [comment]
@ehsantn I'm not sure investing time into doing anything with this now is worth it.  We have one solution already (use y[:]).  I actually kind of like that you get a lowering error if you don't have the "y = np.zeros(3)" line.  It tells you that you are doing something wrong.  If we want to fix this, then we need to find defs in the loop body that are live at exit from the loop and then transform the use of "y" in the loop to y_temp[0] and then after the loop do "y = y_temp[0]".  It isn't the worst thing in the world but I think we have some higher-priority stuff right now?

**Comment by ehsantn** [comment]
@DrTodd13 I agree - this is low priority. I agree with your suggested eventual fix as well.
